app-creator/client/src/widgets/dropzone-widget/dropzone-widget.ts

```diff
@@ -132,15 +131,20 @@ export class Dropzone extends LitElement {
     clone.id += `-${store.getState().widgetIDs[widget.id]}`;
 
     // Return early if the widget has been added to another panel earlier.
-    const template = store.getState().template.widgets;
-    for (const id in template) {
+    const widgets = store.getState().template.widgets;
+    for (const id in widgets) {
       if (
-        typeof template[id] === 'object' &&
-        !Array.isArray(template[id]) &&
+        typeof widgets[id] === 'object' &&
+        !Array.isArray(widgets[id]) &&
         id !== parent!.id &&
-        template[id].children.includes(clone.id)
+        widgets[id].children.includes(clone.id)
       ) {
-        return;
+        const widget = widgets[clone.id].widgetRef;
+        const draggable = widget.parentElement as DraggableWidget;
+        const dropzone = draggable.parentElement as Dropzone;
+        dropzone.removeChild(draggable);
```

Comment:
Is it possible to do the removal outside of the `getChildrenIds()` method? `getChildrenIds()` should ideally only return the IDs and not produce any side effects.

Reply:
The `getChildrenIds` method only has this, everything else is outside.

```Typescript
 /**
   * Return children IDs for a given dropzone.
   */
  getChildrenIds(): string[] {
    return Array.from(this.children).map((el) => {
      return el.firstElementChild!.id;
    });
  }
```

Reply:
I think it looked like its all inside because the code between was collapsed

![Screenshot 2020-08-03 at 5 26 15 PM](https://user-images.githubusercontent.com/26859947/89228975-95a70c00-d5ae-11ea-8dbf-25210d79ad50.png)

Reply:
Yes I totally missed the collapsed part, my bad. :)